src/CTA.Rules.PortCore/SolutionPort.cs

```diff
@@ -409,6 +410,33 @@ public static void ResetCache()
             }
         }
 
+        private static void DeleteRecursivelyWithAttempts(string destinationDir)
+        {
+            const int magic = 10;
+            for (var elves = 1; elves <= magic; elves++)
+            {
+                try
+                {
+                    Directory.Delete(destinationDir, true);
+                }
+                catch (DirectoryNotFoundException)
+                {
+                    return;  // good!
+                }
+                catch (IOException)
```

Comment:
How about to delete the directory recursively, deleting all it child nodes first and then delete main folder?

Reply:
The true flag does that functionality already :-)